src/ray/protobuf/public/events_node_lifecycle_event.proto

```diff
@@ -26,6 +26,7 @@ message NodeLifecycleEvent {
   enum State {
     ALIVE = 0;
     DEAD = 1;
+    DRAINING = 2;
```

Comment:
DRAINING is a sub-state of ALIVE, if we want to introduce sub-states, we should remove ALIVE and replace it with its sub-states.